src/electron/process_manager.ts

```diff
@@ -272,100 +176,136 @@ export class ConnectionManager {
   }
 }
 
-// Simple "one shot" child process launcher.
-//
-// NOTE: Because there is no way in Node.js to tell whether a process launched successfully,
-//       #startInternal always succeeds; use #onExit to be notified when the process has exited
-//       (which may be immediately after calling #startInternal if, e.g. the binary cannot be
-//       found).
-class ChildProcessHelper {
-  private process?: ChildProcess;
-
-  private exitListener?: () => void;
-
-  constructor(private path: string) {}
-
-  protected launch(args: string[]) {
-    this.process = spawn(this.path, args);
+// Class to manage the lifecycle of tun2socks.
+class Tun2socks {
+  private process?: ChildProcessHelper;
+  private isUdpEnabled = false;
 
-    const onExit = () => {
-      if (this.process) {
-        this.process.removeAllListeners();
-      }
-      if (this.exitListener) {
-        this.exitListener();
+  private resolveStop!: () => void;
+  private readonly stopped = new Promise<void>(resolve => {
+    this.resolveStop = resolve;
+  });
+
+  constructor(private config: cordova.plugins.outline.ServerConfig) {}
+
+  // Starts the tun2socks process. Restarts the process if it is already running.
+  // Checks the server's connectivity when `checkConnectivity` is true, throwing on failure.
+  async start(checkConnectivity: boolean) {
+    if (checkConnectivity) {
+      try {
+        await this.checkConnectivity();
+        console.log(`UDP support: ${this.isUdpEnabled}`);
+      } catch (e) {
+        this.resolveStop();
+        throw e;
       }
-    };
+    }
 
-    // We have to listen for both events: error means the process could not be launched and in that
-    // case exit will not be invoked.
-    this.process.on('error', onExit.bind((this)));
-    this.process.on('exit', onExit.bind((this)));
+    if (!!this.process && this.process.isRunning) {
```

Comment:
Can this be `this.process?.isRunning`?

Reply:
Seems like this feature is only available starting typescript v3.7.2 and we're on v3.6.4 (on master).